### PR TITLE
Expose maxMethodsCount to config

### LINKED ISSUE #82: Increase maximum methods count
Hello, thank you for developing such a great tool.

It seems like there's a limit of 100,000 methods. I'm wondering if there are any plans to increase this limit. In my case, I'm trying to analyze a project with approximately 6 million methods, but I'm encountering issues where the scavenger is disabled due to the maximum methods count.

### LINKED ISSUE #86: UNKNOWN: HTTP status code 413 / Request Entity Too Large
프로젝트에 적용할 때 다음과 같은 에러가 발생합니다. Request의 grpc limit이 걸리는 것 같습니다.

```
ERROR [my-service,,] [] [my-service-f66875bd8-7rzlb:2023-08-16 11:19:13.861] [scavenger scheduler] [c.n.s.j.s.Scheduler:publishCodeBaseIfNeeded:180] [scavenger] codebase publish failed
sc.io.grpc.StatusRuntimeException: UNKNOWN: HTTP status code 413
invalid content-type: text/plain; charset=utf-8
headers: Metadata(:status=413,content-type=text/plain; charset=utf-8,content-length=50,date=Wed, 16 Aug 2023 02:19:10 GMT,server=envoy)
DATA-----------------------------
Status: 413
Description: Request Entity Too Large
```

규모가 큰 프로젝트에서도 사용할 수 있도록 API, Collector 서버의 리소스가 충분하다면 동작 가능하게 구현을 부탁드립니다. 감사합니다.

### PR DESCRIPTION
**Pull request by kojandy**
Resolves #82
Resolves #86

### CODE REVIEW

File: scavenger-agent-java/src/main/java/com/navercorp/scavenger/javaagent/ScavengerAgent.java [hunk 1]
```diff
@@ -39,43 +39,18 @@ public static void premain(String args, Instrumentation inst) {
 
         new ScavengerBanner(config).printBanner(System.out);
 
-        CodeBase codeBase = null;
+        Scheduler scheduler = new Scheduler(config);
```
Comment (taeyeon-Kim) [inline scavenger-agent-java/src/main/java/com/navercorp/scavenger/javaagent/ScavengerAgent.java:42]: The codebase scan only runs the first time, and it's awkward to have this task delegated to a scheduler object. 
Is there a reason for moving it to the scheduler?

Comment (kojandy) [reply scavenger-agent-java/src/main/java/com/navercorp/scavenger/javaagent/ScavengerAgent.java:42]: If scanning or uploading codebase fails, the scheduler will schedule a retry of the task. (only when async mode is off)

File: scavenger-agent-java/src/main/java/com/navercorp/scavenger/javaagent/model/Config.java [hunk 2]
```diff
@@ -47,6 +47,7 @@ public class Config {
     private int schedulerInitialDelayMillis = 10_000;
     private int schedulerIntervalMillis = 10_000;
     private Integer forceIntervalSeconds;
+    private Integer maxMethodsCount;
```
[the remark this answers is not in the document]
Comment (kojandy) [reply scavenger-agent-java/src/main/java/com/navercorp/scavenger/javaagent/model/Config.java:50]: Updated on 79a8b38.

File: scavenger-agent-java/src/main/java/com/navercorp/scavenger/javaagent/ScavengerAgent.java [hunk 3]
```diff
@@ -39,43 +39,18 @@ public static void premain(String args, Instrumentation inst) {
...
         if (!config.isAsyncCodeBaseScanMode()) {
-            codeBase = scanCodeBase(config);
+            boolean isScanSuccessful = scheduler.scanCodeBase();
```
Comment (taeyeon-Kim) [inline scavenger-agent-java/src/main/java/com/navercorp/scavenger/javaagent/ScavengerAgent.java:44]: It's a small thing, but how about `scanSuccessful`?

**Review by taeyeon-Kim — APPROVED** [review]

File: scavenger-agent-java/src/main/java/com/navercorp/scavenger/javaagent/ScavengerAgent.java (hunk 3 above)
Comment (kojandy) [reply scavenger-agent-java/src/main/java/com/navercorp/scavenger/javaagent/ScavengerAgent.java]: Fixed on d75c57c.

**Review by sohyun-ku — APPROVED** [review]

**Pull request merged**